File: LeetCode Practice/Javascript_Practice/Evaluate Reverse Polish Notation.js
/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function (tokens) {
    let numbers = [];
    let n = tokens.length;
    let symbols = ["+", "-", "/", "*"];

    for (let i = 0; i < n; i++) {
        if (symbols.includes(tokens[i])) {
            let x = numbers.pop();
            let result = 0;

            if (tokens[i] == "+") {
                result = numbers.pop() + x;
            } else if (tokens[i] == "-") {
                result = numbers.pop() - x;
            } else if (tokens[i] == "/") {
                result = Math.trunc(numbers.pop() / x);
            } else {
                result = numbers.pop() * x;
            }

            numbers.push(result);
        } else {
            numbers.push(parseInt(tokens[i]));
        }
    }

    return numbers[0];
};

File: LeetCode Practice/Python_Practive/Evaluate Reverse Polish Notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        numbers = []

        for i, value in enumerate(tokens):
            if (value in ("+", "/", "-", "*")):
                x = numbers.pop()
                
                if value == "+":
                    result = numbers.pop() + x
                elif value == "/":
                    result = int(numbers.pop() / x)
                elif value == "-":
                    result = numbers.pop() - x
                else:
                    result = numbers.pop() * x

                numbers.append(result)
            else:
                numbers.append(int(value))
        
        return numbers[0]
        